vocode/streaming/agent/state_agent.py

```diff
@@ -1366,6 +1398,8 @@ async def call_ai(self, prompt, tool=None, stop=None, stream_output=True):
         streamed = False
         start_time = time.time()
 
+        messages_to_add = []
```

Comment:
Main change in call ai. Record the messages we produce in `produce_interruptible_agent_response_event_nonblocking`

Return them to add to the json transcript